### ISSUE TITLE
Add the possibility to import Types from global loaded Librarys

### ISSUE THREAD

**Issue by jogibear9988** [microsoft/TypeScript#57019]
### 🔍 Search Terms

 import types global 

### ✅ Viability Checklist

- [X] This wouldn't be a breaking change in existing TypeScript/JavaScript code
- [X] This wouldn't change the runtime behavior of existing JavaScript code
- [X] This could be implemented without emitting different JS based on the types of the expressions
- [X] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, new syntax sugar for JS, etc.)
- [X] This isn't a request to add a new utility type: https://github.com/microsoft/TypeScript/wiki/No-New-Utility-Types
- [X] This feature would agree with the rest of our Design Goals: https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals

### ⭐ Suggestion

In browser environemnt you often need to load librarys, wich are developed as ESM modules ans with ESM types globaly, cause they are only delivered as CJS, for example typescript, see: https://github.com/microsoft/TypeScript/issues/57010

There should be a way, so I can import typescript and use it, without loading the library (like import type), but there should be no error if I use instances from the import.

for example at the moment it is like this:

     import type ts form 'typescript'
     const a = ts.SyntaxKind.TaggedTemplateExpression

will error cause the import is only imported as type

maybe something like this:

     import global type ts form 'typescript'
     const a = ts.SyntaxKind.TaggedTemplateExpression

or 

     import global ts form 'typescript'
     const a = ts.SyntaxKind.TaggedTemplateExpression

would be possible?

### 📃 Motivating Example

aa

### 💻 Use Cases

1. What do you want to use this for?
2. What shortcomings exist with current approaches?
3. What workarounds are you using in the meantime?

**Comment by RyanCavanaugh** [comment]
I don't understand what you're asking for. What makes this import "global" ? What's the emitted code supposed to be? If you have a CJS library, how are you loading it from a browser without using a bundler?

**Comment by fatcerberus** [comment]
I’m guessing they’re loading the TS API globally (using a `<script>` tag) and want those globals to be visible at compile time, rather than resolving it as a module.

**Comment by jakebailey** [comment]
If so, then that was #53570, which we decided not to do.

**Comment by jogibear9988** [comment]
> I’m guessing they’re loading the TS API globally (using a `<script>` tag) and want those globals to be visible at compile time, rather than resolving it as a module.

I'm developing my application without using any bundler as a Project wich is using ESM modules. So I tried to imporrt typescript (or also other packages wich are only available as CJS), but it will fail. So I load these modules globaly (via a scrit tag in my main html file), but then I can not use the tyes in my ts file. This is the issue. I can then use "import type", but then I can use the import only as a type definition, but I also need to access objects from it. This will work at runtime since the file is loaded globally

**Comment by jogibear9988** [comment]
I would love to only use ESM, but also TS is not available as a ESM build

**Comment by jogibear9988** [comment]
> I don't understand what you're asking for. What makes this import "global" ? What's the emitted code supposed to be? If you have a CJS library, how are you loading it from a browser without using a bundler?

the emitted code should be the same as with import type, remove the import, leave the rest.

what is the typescript npm packge? is it not CJS? I can load this directly via scrit tag in browser, and I can import it in vscode in my file, but the import will only work if I use a bundler

**Comment by RyanCavanaugh** [comment]
typescript.js creates a `ts` global and, if running in a CJS context, exports that. So what you need is just a .d.ts file that says
```
import typescript = require('typescript');

declare global {
    export import ts = typescript;
}
```
to represent that that's happening

**Comment by jogibear9988** [comment]
@RyanCavanaugh 

I get this error when I try this:

<img width="925" alt="image" src="https://github.com/microsoft/TypeScript/assets/364896/92b0f3ee-f16e-428b-88b7-7d7dc1be34a1">

**Comment by jogibear9988** [comment]
Is this now a question? Cause I can not get this to work, or is this an issue?

**Comment by DanielRosenwasser** [comment]
I guess one thing you could do is have a separate file like `ts-globalizer.d.ts` in your project that looks like this:

```ts
export * from "typescript";

export as namespace ts;
```

**Comment by DanielRosenwasser** [comment]
Though I guess that gives you this error...

> Module '"typescript"' uses 'export =' and cannot be used with 'export *'.

**Comment by DanielRosenwasser** [comment]
Duh:

```ts
import typescript = require("typescript");
export = typescript;

export as namespace ts;
```

[Playground Link](https://www.typescriptlang.org/play?moduleResolution=99&target=99&module=1&filetype=d.ts#code/JYWwDg9gTgLgBDAnmApgZwMZWGeBeOKFARwFdgiAKAIiVU212oEoBuAKBQA9JY4C66LDhgdOPaPACGaOADspIdGCkYUCNKyA)

**Comment by jogibear9988** [comment]
Will also not work:

<img width="796" alt="image" src="https://github.com/microsoft/TypeScript/assets/364896/91c165af-8994-46d4-8736-670ac4118aa0">

**Comment by DanielRosenwasser** [comment]
It's a safety measure - there's `allowUmdGlobalAccess`. @RyanCavanaugh has not tried to mow that one down [in recent deprecations](https://github.com/microsoft/TypeScript/issues/51909). 😄

**Comment by jogibear9988** [comment]
Isn't then my suggestion a valid feature request?

To use a global JS library with type information, I need to create an extra file, create a bunch of code, add an compiler switch wich may be deprecated at some time...

**Comment by DanielRosenwasser** [comment]
It is, and we can keep it open - I just realistically don't know how much traction it will get, versus how much the cost is of adding new syntax and supporting it. The workaround isn't great, but it is totally possible, and the number of secretly-global modules is going to get rarer.

**Comment by jogibear9988** [comment]
Yeah, thanks for the solution. Already tested it, and works nicely.

**Comment by whzx5byb** [comment]
What about this one?
```ts
import ts = require('typescript');

declare global {
    namespace globalThis {
        export { ts }
    }
}
```

**Comment by jogibear9988** [comment]
> ```ts
> import ts = require('typescript');
> 
> declare global {
>     namespace globalThis {
>         export { ts }
>     }
> }
> ```

Yeah, also works. And for me, easier to understand
